### PR TITLE
c2c-gateway: refactoring

### LINKED ISSUE #380: Support JWT tokens on devices
### What is the purpose of the feature?
The current version of the plgd cloud authorizes device connections using opaque tokens. By using JWT tokens, service tokens between plgd Gateways, RA and RD could be dropped and communication between CoAP Gateway and OAuth2.0 Server optimized.

### What specific challenge does this feature solve?
- reduce deployment complexity by not requiring OAuth 2.0 Service clients for plgd services
- reduce the number of requests towards an OAuth 2.0 Server from the CoAP Gateway, introspecting each opaque token received from all connected devices

### What alternatives did I consider?
None

### PR DESCRIPTION
**Pull request by Danielius1922**
- update configuration to be loadable from yaml

### CODE REVIEW

File: cloud2cloud-gateway/service/config.go [hunk 1]
```diff
@@ -1,31 +1,173 @@
...
+	Storage           StorageConfig           `yaml:"storage" json:"storage"`
+	Subscription      SubscriptionConfig      `yaml:"subscription" json:"subscription"`
+}
+
+func (c *ClientsConfig) Validate() error {
+	if err := c.Eventbus.Validate(); err != nil {
+		return fmt.Errorf("eventBus.%w", err)
+	}
+	if err := c.ResourceAggregate.Validate(); err != nil {
+		return fmt.Errorf("resourceDirectory.%w", err)
```
Comment (Danielius1922) [inline cloud2cloud-gateway/service/config.go]: resourceAggregate

File: cloud2cloud-gateway/service/service.go [hunk 2]
```diff
@@ -2,157 +2,291 @@ package service
...
+	}
+	fl.AddFunc(func() {
+		if err := store.Close(ctx); err != nil {
+			log.Errorf("failed to close subscription store: %w", err)
+		}
+	})
+
+	subMgr := NewSubscriptionManager(ctx, store, gwClient, cfg.Clients.Subscription.HTTP.ReconnectInterval, emitEvent)
+	if err := subMgr.LoadSubscriptions(); err != nil {
+		return nil, nil, fmt.Errorf("cannot load subscriptions: %w", err)
```
Comment (jkralik) [inline cloud2cloud-gateway/service/service.go:188]: pls, add `fl.Close`

File: cloud2cloud-gateway/config.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+log:
+  debug: false
+apis:
+  http:
+    address: "0.0.0.0:9100"
+    ownerClaim: "sub"
```
Comment (jkralik) [inline cloud2cloud-gateway/config.yaml]: I expected it under `clients.authorizationServer.ownerClaim`

File: cloud2cloud-gateway/service/service.go (hunk 2 above)
Comment (Danielius1922) [reply cloud2cloud-gateway/service/service.go:188]: Fixed

File: cloud2cloud-gateway/config.yaml (hunk 3 above)
Comment (Danielius1922) [reply cloud2cloud-gateway/config.yaml]: It's not needed at all after I've removed the `CtxWithOwner` calls.

File: cloud2cloud-gateway/service/subscriptions.go [hunk 4]
```diff
@@ -219,39 +271,36 @@ func (s *SubscriptionData) createDeviceSubscription(ctx context.Context, emitEve
...
-		}
+		createSubscriptionFunc = s.createResourceSubscription
 	default:
 		return fmt.Errorf("unsupported subscription type: %v", store.Type_Device)
 	}
+
+	sub, err := createSubscriptionFunc(ctx, emitEvent, &h)
+	if err != nil {
+		// TODO: ak err je grpc unauthenticated -> cancel subscription
+		// test-case?
```
Comment (jkralik) [inline cloud2cloud-gateway/service/subscriptions.go]: TODO

**Review by ondrejtomcik — APPROVED** [review]

**Review by jkralik — APPROVED** [review]

**Pull request merged**